app/code/Kogut/ProductsGeneration/Ui/Component/Listing/GenerateProductsButton.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kogut\ProductsGeneration\Ui\Component\Listing;
+
+use Kogut\ProductsGeneration\Model\Config\Settings;
+use Magento\Framework\View\Element\UiComponent\Control\ButtonProviderInterface;
+use Magento\Framework\UrlInterface;
+
+/**
+ * Generate Products Button
+ */
+class GenerateProductsButton implements ButtonProviderInterface
+{
+    const XML_PATH_TO_PRODUCT_GENERATION_CONTROLLER = 'generate_products/products/generate';
+
+    /**
+     * URL builder
+     *
+     * @var UrlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * @var Settings
+     */
+    private $config;
+
+    /**
+     * @param UrlInterface $urlBuilder
+     */
+    public function __construct(
+        UrlInterface $urlBuilder,
+        Settings $config
+    ) {
+        $this->urlBuilder = $urlBuilder;
+        $this->config = $config;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getButtonData()
+    {
+        $message = sprintf(
+            __("Are you sure you want to generate %s products?")->render(),
```

Comment:
You dont need sprintf here
```
$message = __("Are you sure you want to generate %1 products?", $this->config->getProductsQtyToGenerate())->render()
```
Also change translation to use %1 as parameter

Reply:
Fixed